homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -128,6 +65,25 @@ def __init__(self, name, egardiasystem, rs_enabled=False, rs_codes=None):
             self._rs_codes = rs_codes[0]
         else:
             self._rs_codes = rs_codes
+        self._rs_port = rs_port
+
+        # configure egardia server, including callback
+        if self._rs_enabled:
+            # Set up the egardia server
+            _LOGGER.info("Setting up EgardiaServer")
+            try:
+                if D_EGARDIASRV not in hass.data:
+                    server = egardiaserver.EgardiaServer('', rs_port)
```

Comment:
Why does the server setuo need to be done in the entity? Either do it in setup_platform or in the component setup.

Reply:
I'd probably put it in the component setup, since it's one server per component, not one per platform. At some point you might want to use it for other platforms.

Does the egardia alarm not support giving push feedback for the binary sensors?

Reply:
I mean the egardia component, ie components/egardia.py.

Reply:
Wait, I had this in the generic component set-up. Did you not ask me to move this to the entity (alarm_control_panel)? I am confused now.
The generic component does not provide push feedback for the binary sensors unfortunately - we will need to poll the status.

Reply:
Is there by the way a way to make the polling speed configurable to the user?

Reply:
Yeah, I'm sorry I haven't been very clear in my comments here.

I think you can separate the setup and start of the server, and the registration of the callback.

Setup and start of the server can happen in the component setup function. Registration of callbacks should happen in the entity method `async_added_to_hass`.

I can give an example of that when I'm at a computer. I don't have so much time now. I'm traveling and have meetings.

Reply:
I really appreciate all your effort! I will give it another try and hopefully it is what you meant.